### PR TITLE
Update dex to handle market routes

### PR DESCRIPTION
**Pull request by eddieharrison**
## Features
* Allows a direct url to be shared for each market pair (e.g. https://dex.aphelion.org/#/app/trade/NEO-NEX)
* Allows forwards/backwards navigating in the browser between market pairs

## Changes
* Dex component loads currentMarket from the /trade/:market route param
* Handles invalid market (e.g. "/app/trade/invalid") - redirects to default market: NEO-APH
* Updated megamenu to change the route only
* Added beforeRouteUpdate method to setCurrentMarket when route/market changes
* Updated sidebar to point "Trade DEX" to /app/trade/NEO-APH

## Notes
* Please pull and test if possible first
* Also, think of anything I may not be aware of?
* Uses fetchMarkets data for validation so new markets will automatically work in the future

### CODE REVIEW

File: src/services/dex.js [hunk 1]
```diff
@@ -2888,4 +2888,19 @@ export default {
     // console.log(`${JSON.stringify(res)}`);
     return res.data.kycStatus;
   },
+
+  getMarketDataFromName(marketName) {
+    let marketData = false;
+    store.state.markets.forEach((market) => {
```
Comment (jsolman) [inline src/services/dex.js]: Could use lodash here for brevity `_.find`

File: src/services/dex.js [hunk 2]
```diff
@@ -2888,4 +2888,19 @@ export default {
...
+      if (market.marketName === marketName) {
+        marketData = market;
+      }
+    });
+    return marketData;
+  },
+
+  getDefaultRoute() {
+    const defaultSettings = this.$constants.defaultSettings;
+    return defaultSettings.DEX_BASE_ROUTE + defaultSettings.DEX_MARKET_PAIR;
```
Comment (jsolman) [inline src/services/dex.js]: `defaultSetings.LANDING_ROUTE`?

File: src/components/Sidebar.vue [hunk 3]
```diff
@@ -17,7 +17,7 @@
           </span>
           <span class="label">{{ $t('dashboard') }}</span>
         </router-link>
-        <router-link to="/app/dex">
+        <router-link to="/app/trade/NEO-APH">
```
Comment (jsolman) [inline src/components/Sidebar.vue]: use `LANDING_ROUTE` constant?

File: src/services/dex.js (hunk 1 above)
Comment (eddieharrison) [reply src/services/dex.js]: Updated to use lodash.

File: src/services/dex.js (hunk 2 above)
Comment (eddieharrison) [reply src/services/dex.js]: getDefaultRoute() removed, constant LANDING_ROUTE replaced.

File: src/components/Sidebar.vue (hunk 3 above)
Comment (eddieharrison) [reply src/components/Sidebar.vue]: Done.

**Review by jsolman — APPROVED** [review]

**Pull request merged**